controllers/cluster_controller.go

```diff
@@ -51,13 +53,46 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	}
 	ctx = loggermeta.NewContext(ctx, meta)
 
-	// your logic here
+	cluster := &capiv1alpha3.Cluster{}
+	err := r.Get(ctx, req.NamespacedName, cluster)
+	if err != nil {
+		return ctrl.Result{}, microerror.Mask(err)
+	}
 
-	return ctrl.Result{}, nil
+	// Based on https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machine_controller.go.
+	if !cluster.DeletionTimestamp.IsZero() {
+		res, err := r.reconcileDelete(ctx, cluster)
+		if err != nil {
+			requeueAfter := 30 * time.Second
+			r.Log.Errorf(ctx, err, "failed to reconcile, requeuing after %#q", requeueAfter)
+			return ctrl.Result{RequeueAfter: requeueAfter}, nil
```

Comment:
Errors are handled here and the request is re-queued after 30s. This is quite naive approach but I don't want to spend too much time and have easy way to debug. Using standard logger will print you the panic-style-stack from the zap logger invocation which is somewhere in controller-runtime library. Not very useful. In the future we can try to implement something more sophisticated. I'll also talk with upstream guys about that.